### ISSUE TITLE
About reproducing the results

### ISSUE THREAD

**Issue by yiqings** [Yura52/tabular-dl-revisiting-models#28]
Hi there,

Thanks for the wonderful codes and insightful paper "Revisiting Deep Learning Models for Tabular Data". 

---
We (@sxjscience) are trying to add FT_Transformer into [AutoMM](https://github.com/awslabs/autogluon/tree/master/text/src/autogluon/text/automm) in [autogluon](https://github.com/awslabs/autogluon). And reproduce the results of FT_Transformer (i.e.Table 2 in [1]), where the training and network configurations are set to identically the same as the original paper (i.e. Table 12 in [1]) :
```
learning_rate: 1.0e-4
weight_decay: 1.0e-5
patience: 16
```
We follow this configurations consistently for all datasets.

----

We found that 
- (1) For smaller datasets, similar or marginally better results can be achieved. 
- (2) For larger datasets (YA,MI), a performance gap exists. We also found that for larger datasets, the FT_Transformer didn't converge with `iteration=100`, hence we use `epoch=100`.

Our re-implemented and reproduced results (https://github.com/awslabs/autogluon/blob/master/examples/automm/README.md) are as follows:

Datasets | ca | ad | he | ja | hi | al | ep | ye | co | ya | ml 
----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  
metrics | rmse | acc | acc | acc | acc | acc | acc | rmse | acc | rmse | rmse
problem_type | regression | binary | multiclass | multiclass | binary | multiclass | binary | regression | multiclass | regression | regression
#objects | 20640 | 48842 | 65196 | 83733 | 98050 | 108000 | 500000 | 515345 | 581012 | 709877 | 1200192
#num. | 8 | 6 | 27 | 54 | 28 | 128 | 2000 | 90 | 54 | 699 | 136
#cat. | 0 | 8 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0
#classes | - | 2 | 100 | 4 | 2 | 1000 | 2 | - | 7 | - | -
Best in [1] | 0.459 | 0.859 | 0.396 | 0.732 | 0.729 | 0.963 | 0.8982 | 8.794 | 0.970 | 0.753 | 0.745
FT-Transformer in [1] | 0.459 | 0.859 | 0.391 | 0.732 | 0.729 | 0.960 | 0.8982 | 8.855 | 0.970 | 0.756 | 0.746
AutoMM FT-Transformer + Const Lr=1e-4 | 0.404 | 0.863 | 0.388 | 0.727 | 0.729 | 0.954 | OverflowError | 0.781 | 0.964 | AttributeError | 0.904
AutoMM FT-Transformer + Cosine Lr=1e-4| 0.401 | 0.864 | 0.388 | 0.728 | 0.729 | 0.952 | OverflowError | 0.780 | 0.964 | 0.765 | 0.924

---

Hence, we are curious about whether are any special settings or tricks for lager datasets (YA,MI)? 

Thanks.

---
### Reference
[1] "Revisiting Deep Learning Models for Tabular Data"

**Comment by sxjscience** [comment]
@Yura52 Would like to have your guidance on this issue. Thanks!

**Comment by Yura52** [comment]
Hi! @yiqings @sxjscience

I'm glad that you have found the paper useful! Also, many thanks for such a detailed report! This is a highly valuable type of feedback for us.

No, there are no special tricks applied to big datasets. Overall, the difference in results indicate that there is difference in hyperparameters, protocols or implementations. Before diving into details, please, could you clarify what results you are trying to reproduce?

- on the one hand, you mention Table 2. In that table, for different datasets, there are different hyperparameters, which are tuned separately for each dataset. For example, you can compare the configs for the [California Housing](https://github.com/Yura52/tabular-dl-revisiting-models/blob/f65a93329a58c80ea486a212feeba6babf9fdd4f/output/california_housing/ft_transformer/tuned/0.toml) and [Covertype](https://github.com/Yura52/tabular-dl-revisiting-models/blob/f65a93329a58c80ea486a212feeba6babf9fdd4f/output/covtype/ft_transformer/tuned/0.toml) datasets.
- on the other hand, you mention Table 12. This is the "default" configuration for FT-Transformer. The results for the default model are reported in Table 4 (note that the results are provided for **ensembles**, though, you can find the results for single models in this repository; for example, see this folder for the [California Housing](https://github.com/Yura52/tabular-dl-revisiting-models/tree/f65a93329a58c80ea486a212feeba6babf9fdd4f/output/california_housing/ft_transformer/default) dataset; 15 results, since the experiment was run with 15 different random seeds).

P.S. It is "MI" (the first two letters of "Microsoft"), not "ML" :)

**Comment by Yura52** [comment]
@yiqings great!

Btw, what metric are you using for the regression problems? The difference with our results is too large, especially on the YE dataset. My hypothesis is that you normalize the regression target and calculate the metric without converting the target and predictions to the original units (as we do [here](https://github.com/Yura52/tabular-dl-revisiting-models/blob/f65a93329a58c80ea486a212feeba6babf9fdd4f/lib/metrics.py#L22)). Is it correct?

**Comment by yiqings** [comment]
Hi @Yura52,

Thanks for notifying the issue in metric computation. Yes, that's very correct: We performed normalizations for all features (input and targets), and calculated the RMSE at scale without converting it to the origin scale. 

We will put more details to [our reproduced results](https://github.com/awslabs/autogluon/blob/master/examples/automm/README.md) very soon.   

Thanks once again for your reply!

**Comment by Yura52** [comment]
@yiqings are you using the latest version (v2) of the paper? Here is the link:
https://arxiv.org/pdf/2106.11959.pdf

Asking because you are mentioning Table 1 which describes dataset properties, not results.

**Comment by yiqings** [comment]
@Yura52 Yes, thanks for the input. 

Table 1 is about the dataset, and actually we are comparing with the Table 2. 

We have updated the reproduced results in the last row, as follows:

Datasets | ca | ad | he | ja | hi | al | ep | ye | co | ya | mi 
----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  | ----  
metrics | rmse | acc | acc | acc | acc | acc | acc | rmse | acc | rmse | rmse
problem_type | regression | binary | multiclass | multiclass | binary | multiclass | binary | regression | multiclass | regression | regression
#objects | 20640 | 48842 | 65196 | 83733 | 98050 | 108000 | 500000 | 515345 | 581012 | 709877 | 1200192
#num. | 8 | 6 | 27 | 54 | 28 | 128 | 2000 | 90 | 54 | 699 | 136
#cat. | 0 | 8 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0
#classes | - | 2 | 100 | 4 | 2 | 1000 | 2 | - | 7 | - | -
Best in [1] | 0.459 | 0.859 | 0.396 | 0.732 | 0.729 | 0.963 | 0.8982 | 8.794 | 0.970 | 0.753 | 0.745
FT-Transformer in [1] | 0.459 | 0.859 | 0.391 | 0.732 | 0.729 | 0.960 | 0.8982 | 8.855 | 0.970 | 0.756 | 0.746
AutoMM FT-Transformer | 0.482 | 0.859 | 0.379 | 0.721 | 0.726 | 0.949 | RuntimeError | 8.891 | 0.963 | 0.769 | 0.761

We use a consistent setting, so there is a performance gap. But they have approach the same scale. This gap can be closed with an identical model configurations.

[1] Revisiting Deep Learning Models for Tabular Data, 2021, https://arxiv.org/pdf/2106.11959.pdf

**Comment by yiqings** [comment]
Thanks you once again for your helpful guidance @Yura52 ! And our questions have been well addressed with you help.
